applications/TrilinosApplication/custom_python/add_trilinos_strategies_to_python.cpp

```diff
@@ -114,29 +117,34 @@ void AddStrategies(pybind11::module& m)
-    .def(py::init< ModelPart&, bool >())
-    .def("Predict", &TrilinosBaseSolvingStrategyType::Predict)
-    .def("Initialize", &TrilinosBaseSolvingStrategyType::Initialize)
-    .def("Solve", &TrilinosBaseSolvingStrategyType::Solve)
-    .def("IsConverged", &TrilinosBaseSolvingStrategyType::IsConverged)
-    .def("CalculateOutputData", &TrilinosBaseSolvingStrategyType::CalculateOutputData)
-    .def("SetEchoLevel", &TrilinosBaseSolvingStrategyType::SetEchoLevel)
-    .def("GetEchoLevel", &TrilinosBaseSolvingStrategyType::GetEchoLevel)
-    .def("SetRebuildLevel", &TrilinosBaseSolvingStrategyType::SetRebuildLevel)
-    .def("GetRebuildLevel", &TrilinosBaseSolvingStrategyType::GetRebuildLevel)
-    .def("SetMoveMeshFlag", &TrilinosBaseSolvingStrategyType::SetMoveMeshFlag)
-    .def("MoveMeshFlag", &TrilinosBaseSolvingStrategyType::MoveMeshFlag)
-    .def("MoveMesh", &TrilinosBaseSolvingStrategyType::MoveMesh)
-    .def("Clear", &TrilinosBaseSolvingStrategyType::Clear)
-    .def("Check", &TrilinosBaseSolvingStrategyType::Check)
-    .def("InitializeSolutionStep", &TrilinosBaseSolvingStrategyType::InitializeSolutionStep)
-    .def("FinalizeSolutionStep", &TrilinosBaseSolvingStrategyType::FinalizeSolutionStep)
-    .def("SolveSolutionStep", &TrilinosBaseSolvingStrategyType::SolveSolutionStep)
-    .def("GetModelPart", &TrilinosBaseSolvingStrategyType::GetModelPart)
-    ;
+        .def(py::init< ModelPart&, bool >())
+        .def("Predict", &TrilinosBaseSolvingStrategyType::Predict)
+        .def("Initialize", &TrilinosBaseSolvingStrategyType::Initialize)
+        .def("Solve", &TrilinosBaseSolvingStrategyType::Solve)
+        .def("IsConverged", &TrilinosBaseSolvingStrategyType::IsConverged)
+        .def("CalculateOutputData", &TrilinosBaseSolvingStrategyType::CalculateOutputData)
+        .def("SetEchoLevel", &TrilinosBaseSolvingStrategyType::SetEchoLevel)
+        .def("GetEchoLevel", &TrilinosBaseSolvingStrategyType::GetEchoLevel)
+        .def("SetMoveMeshFlag", &TrilinosBaseSolvingStrategyType::SetMoveMeshFlag)
+        .def("MoveMeshFlag", &TrilinosBaseSolvingStrategyType::MoveMeshFlag)
+        .def("MoveMesh", &TrilinosBaseSolvingStrategyType::MoveMesh)
+        .def("Clear", &TrilinosBaseSolvingStrategyType::Clear)
+        .def("Check", &TrilinosBaseSolvingStrategyType::Check)
+        .def("InitializeSolutionStep", &TrilinosBaseSolvingStrategyType::InitializeSolutionStep)
+        .def("FinalizeSolutionStep", &TrilinosBaseSolvingStrategyType::FinalizeSolutionStep)
+        .def("SolveSolutionStep", &TrilinosBaseSolvingStrategyType::SolveSolutionStep)
+        .def("GetModelPart", [](TrilinosBaseSolvingStrategyType& self) -> ModelPart& { return self.GetModelPart(); })
+        ;
+
+    // Implicit strategy base class
+    py::class_< TrilinosImplicitSolvingStrategyType, typename TrilinosImplicitSolvingStrategyType::Pointer, TrilinosBaseSolvingStrategyType >(m, "TrilinosImplicitSolvingStrategy")
```

Comment:
as other comment, please make it consistent with the serial one (future PR is good for me)

Reply:
I think I don't get your point at all. I guess you mean having the very same exposure in serial and Trilinos classes. Am I wrong?

Reply:
I only mean to expose it consistently:

~~~py
SolvingStrategy
ImplicitSolvingStrategy

TrilinosSolvingStrategy
TrilinosImplicitSolvingStrategy
~~~